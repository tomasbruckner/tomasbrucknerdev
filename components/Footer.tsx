import React from 'react';
import { Typography, useTheme } from '@material-ui/core';
import { useTranslation } from 'react-i18next';

const Footer = () => {
  const { t } = useTranslation();
  const theme = useTheme();

  return (
    <footer>
      <div
        style={{
          width: '100%',
          minHeight: '64px',
          color: 'white',
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
          backgroundColor: theme.palette.primary.main,
        }}
      >
        <Typography>{t('footer.copyright')}</Typography>
      </div>
    </footer>
  );
};

export default Footer;
