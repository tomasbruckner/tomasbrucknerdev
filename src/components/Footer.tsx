import React, { FC } from 'react';
import { makeStyles, Typography } from '@material-ui/core';
import { t } from '../utils/i18n';

const useStyles = makeStyles((theme) => ({
  container: {
    width: '100%',
    minHeight: '64px',
    color: theme.palette.common.white,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.palette.primary.main,
  },
}));

const Footer: FC = () => {
  const classes = useStyles();

  return (
    <footer>
      <div className={classes.container}>
        <Typography>{t('footer.copyright')}</Typography>
      </div>
    </footer>
  );
};

export default Footer;
