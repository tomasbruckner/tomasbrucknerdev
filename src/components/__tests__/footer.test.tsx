jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (text: string) => text,
  }),
}));

import React from 'react';
import Footer from 'components/Footer';
import renderer from 'react-test-renderer';

it('Footer renders correctly', () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});
