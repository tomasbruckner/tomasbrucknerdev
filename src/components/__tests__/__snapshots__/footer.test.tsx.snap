// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer>
  <div
    className="makeStyles-container-1"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Tomáš Bruckner © 2018 - 2019
    </p>
  </div>
</footer>
`;
